### ISSUE TITLE
[object detection] retrain model is not effective

### ISSUE THREAD

**Issue by WZMIAOMIAO** [tensorflow/models#7611]
### System information
- **What is the top-level directory of the model you are using: "/home/ubuntu/tensorflowOBDT"
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow) : No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 16.04
- **TensorFlow installed from (source or binary): pip install
- **TensorFlow version (use command below) : b'v1.13.0-rc2-5-g6612da8' 1.13.1 CPU
- **Python version : Python3.7
- **Bazel version: None
- **CUDA/cuDNN version: None
- **GPU model and memory: None
- **Exact command to reproduce:
     PIPELINE_CONFIG_PATH="/xxx/ssd_mobilenet_v2_coco.config"
     MODEL_DIR="/xxx/train/"
     NUM_TRAIN_STEPS=3000
     SAMPLE_1_OF_N_EVAL_EXAMPLES=1
     python object_detection/model_main.py \
         --pipeline_config_path=${PIPELINE_CONFIG_PATH} \
         --model_dir=${MODEL_DIR} \
         --num_train_steps=${NUM_TRAIN_STEPS} \
         --sample_1_of_n_eval_examples=$SAMPLE_1_OF_N_EVAL_EXAMPLES \
         --alsologtostderr

### Describe the problem
When I retrain ssd_moblenet_v2_coco or ssd_resnet_50_fpn_coco with PASCAL VOC2012 datesset, I find Average Precision(coco_detection_metrics) is lower(start with 0.009). I think it‘s not normal. I've used it before. I remember it should start with 0.15~0.25(AP) maybe. I  train 3000 step, AP from 0.009 to 0.20. I guess the model does'n load pretrain parameter.

#### To generate pacal voc datasets, the code is following:
#From tensorflow/models/research/
wget http://host.robots.ox.ac.uk/pascal/VOC/voc2012/VOCtrainval_11-May-2012.tar
tar -xvf VOCtrainval_11-May-2012.tar
python object_detection/dataset_tools/create_pascal_tf_record.py \
    --label_map_path=object_detection/data/pascal_label_map.pbtxt \
    --data_dir=VOCdevkit --year=VOC2012 --set=train \
    --output_path=pascal_train.record
python object_detection/dataset_tools/create_pascal_tf_record.py \
    --label_map_path=object_detection/data/pascal_label_map.pbtxt \
    --data_dir=VOCdevkit --year=VOC2012 --set=val \
    --output_path=pascal_val.record

#### The configuration file I used is from research/object_detection/samples/configs:
model {
  ssd {
    num_classes: 20
    box_coder {
      faster_rcnn_box_coder {
        y_scale: 10.0
        x_scale: 10.0
        height_scale: 5.0
        width_scale: 5.0
      }
    }
    matcher {
      argmax_matcher {
        matched_threshold: 0.5
        unmatched_threshold: 0.5
        ignore_thresholds: false
        negatives_lower_than_unmatched: true
        force_match_for_each_row: true
      }
    }
    similarity_calculator {
      iou_similarity {
      }
    }
    anchor_generator {
      ssd_anchor_generator {
        num_layers: 6
        min_scale: 0.2
        max_scale: 0.95
        aspect_ratios: 1.0
        aspect_ratios: 2.0
        aspect_ratios: 0.5
        aspect_ratios: 3.0
        aspect_ratios: 0.3333
        reduce_boxes_in_lowest_layer: true
      }
    }
    image_resizer {
      fixed_shape_resizer {
        height: 300
        width: 300
      }
    }
    box_predictor {
      convolutional_box_predictor {
        min_depth: 0
        max_depth: 0
        num_layers_before_predictor: 0
        use_dropout: false
        dropout_keep_probability: 0.8
        kernel_size: 3
        box_code_size: 4
        apply_sigmoid_to_scores: false
        conv_hyperparams {
          activation: RELU_6,
          regularizer {
            l2_regularizer {
              weight: 0.00004
            }
          }
          initializer {
            truncated_normal_initializer {
              stddev: 0.03
              mean: 0.0
            }
          }
        }
      }
    }
    feature_extractor {
      type: 'ssd_inception_v2'
      min_depth: 16
      depth_multiplier: 1.0
      conv_hyperparams {
        activation: RELU_6,
        regularizer {
          l2_regularizer {
            weight: 0.00004
          }
        }
        initializer {
          truncated_normal_initializer {
            stddev: 0.03
            mean: 0.0
          }
        }
        batch_norm {
          train: true,
          scale: true,
          center: true,
          decay: 0.9997,
          epsilon: 0.001,
        }
      }
      override_base_feature_extractor_hyperparams: true
    }
    loss {
      classification_loss {
        weighted_sigmoid {
        }
      }
      localization_loss {
        weighted_smooth_l1 {
        }
      }
      hard_example_miner {
        num_hard_examples: 3000
        iou_threshold: 0.99
        loss_type: CLASSIFICATION
        max_negatives_per_positive: 3
        min_negatives_per_image: 0
      }
      classification_weight: 1.0
      localization_weight: 1.0
    }
    normalize_loss_by_num_matches: true
    post_processing {
      batch_non_max_suppression {
        score_threshold: 1e-8
        iou_threshold: 0.6
        max_detections_per_class: 100
        max_total_detections: 100
      }
      score_converter: SIGMOID
    }
  }
}

train_config: {
  batch_size: 24
  optimizer {
    rms_prop_optimizer: {
      learning_rate: {
        exponential_decay_learning_rate {
          initial_learning_rate: 0.004
          decay_steps: 800
          decay_factor: 0.95
        }
      }
      momentum_optimizer_value: 0.9
      decay: 0.9
      epsilon: 1.0
    }
  }
  fine_tune_checkpoint: "/xxx/ssd_mobilenet_v2_coco/model.ckpt"
  fine_tune_checkpoint_type: "detection" 
  #Note: The below line limits the training process to 200K steps, which we
  #empirically found to be sufficient enough to train the pets dataset. This
  #effectively bypasses the learning rate schedule (the learning rate will
  #never decay). Remove the below line to train indefinitely.
  num_steps: 3000
  data_augmentation_options {
    random_horizontal_flip {
    }
  }
  data_augmentation_options {
    ssd_random_crop {
    }
  }
}

train_input_reader: {
  tf_record_input_reader {
    input_path: "/xxx/pascal_train.record"
  }
  label_map_path: "/xxx/pascal_label_map.pbtxt"
}

eval_config: {
  num_examples: 100
  #Note: The below line limits the evaluation process to 10 evaluations.
  #Remove the below line to evaluate indefinitely.
  max_evals: 10
}

eval_input_reader: {
  tf_record_input_reader {
    input_path: "/xxx/pascal_val.record"
  }
  label_map_path: "/xxx/pascal_label_map.pbtxt"
  shuffle: false
  num_readers: 1
}

#### The train code is following:
#From the tensorflow/models/research/ directory
PIPELINE_CONFIG_PATH="/xxx/ssd_mobilenet_v2_coco.config"
MODEL_DIR="/xxx/train/"
NUM_TRAIN_STEPS=3000
SAMPLE_1_OF_N_EVAL_EXAMPLES=1
python object_detection/model_main.py \
    --pipeline_config_path=${PIPELINE_CONFIG_PATH} \
    --model_dir=${MODEL_DIR} \
    --num_train_steps=${NUM_TRAIN_STEPS} \
    --sample_1_of_n_eval_examples=$SAMPLE_1_OF_N_EVAL_EXAMPLES \
    --alsologtostderr

#### Due to computer running tensorflow can not copy info, I summary some information:
some Waring: some function is deprecated.
four Waring: Variable is availble in checkpoint, but  has an incompatible shape with model variable. Checkpoint shape:[[1, 1, 256, 512]], model variable shape: [[3, 3, 256, 512]].
some Instructions for updating.
etc.

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by WZMIAOMIAO** [comment]
@tensorflowbutler I have filled out template.

**Comment by WZMIAOMIAO** [comment]
It is work.

**Comment by SLAMgreen** [comment]
Did you add some parameters in config flles？
